godotproject/AntAttackFire.gd

```diff
@@ -1,24 +1,30 @@
-extends Node2D
+extends KinematicBody2D
 
-var speed = 5
-var velocity = Vector2()
+const FLOOR = Vector2(0, -1)
+const SPEED = 250
+
+var velocity = Vector2(0, 0)
 var distance_traveled = 0
 var direction = 1
+var attackDamage = 1
 
 func _physics_process(_delta):
-	velocity.x = speed * direction
+
+	velocity.x = SPEED * direction
 	velocity.y = 0 # not needed, it's already 0
-	translate(velocity)
-	distance_traveled += speed
+	velocity = move_and_slide(velocity, FLOOR)
+	distance_traveled += SPEED * _delta
 	
-	if distance_traveled > 200:
-		queue_free()
+	if distance_traveled > 200 || is_on_wall():
+		dead()
```

Comment:
Can you add the fireball poof sound here please. Its the same one as in the _on_PlayerCollisionArea_body_entered function. $FireballExplodeSoundPlayer.play(0)